### PR TITLE
Set the default locale for the JVM to English to prevent locale-speci…

### LINKED ISSUE #26438: Keycloak cannot run on windows machine in dev-mode. Because non-English systems cannot support keycloak's package's.
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dependencies

### Describe the bug

image

When i change the system language problem solved. But I think that is not a proper solution.

### Version

23.0.4

### Expected behavior

To start a keycloak dev server.

### Actual behavior

ERROR: No enum constant org.keycloak.common.Profile.ProfileValue.COMMUNİTY

COMMUNITY is not the package that keycloak's looking for.

### How to Reproduce?

When i change system language in windows. Command Prompt's language changed and problem solved. But I think that's not a proper way.

### Anything else?

Thank you for support.

### PR DESCRIPTION
**Pull request by Pepo48**
…fic character variations

* introduced the JAVA_LOCALE variable within kc.sh|bat in order to control JVM locale

Closes #26438

@vmuzikar I decided to preserve the way we handle the other java options in our scripts and I added a little bit of logic around appending the java language options, too.

Here I demonstrate the new default behaviour on a Turkish system compared to the latest release behaviour:
image

And this happens when a user decides to override the new default behaviour (as they are allowed to do so):
image

Previously I ran the tests [here](https://github.com/Pepo48/keycloak/actions/runs/8255969714) - every workflow was green.

Let me know your opinion about this approach and we can eventually ask someone from the core team about their opinion as well.
Thanks.

### CODE REVIEW

**Review by shawkins — APPROVED** [review]
LGTM, does JAVA_LOCALE need documented anywhere?

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks @Pepo48!

There's a small regression risk with this change. In theory, setting english locale in non-english environment where previously a different locale was set by default by JVM, might introduce some side effects. But I think it should be safe enough as we do not really rely on JVM locale in UI localization. @keycloak/ui Can you please confirm?

**Comment by vmuzikar** [comment]
> does JAVA_LOCALE need documented anywhere?

@shawkins I don't think so. It's just an escape hatch, normally users should not need to touch it at all.

**Comment by ssilvert** [comment]
> LGTM, thanks @Pepo48!
> 
> There's a small regression risk with this change. In theory, setting english locale in non-english environment where previously a different locale was set by default by JVM, might introduce some side effects. But I think it should be safe enough as we do not really rely on JVM locale in UI localization. @keycloak/ui Can you please confirm?

UI localization will depend on the locale setting of the user who logged in.  That locale is calculated on the back end.  So basically, the UI does whatever the back end tells it to do.  

@vmuzikar Does that answer your question?

**Comment by vmuzikar** [comment]
> @vmuzikar Does that answer your question?

Let me rephrase. :) Do we rely on default system locale in UI anywhere? E.g. when a realm or user locale is not specified and no locale is present in request headers.

**Comment by vmuzikar** [comment]
> Do we rely on default system locale in UI anywhere? E.g. when a realm or user locale is not specified and no locale is present in request headers.

@ssilvert Ok, I think I can answer that. :) We fallback to English:
https://github.com/keycloak/keycloak/blob/0fb6bdfcac5139b47176215c6033b5c5fad9a3d2/services/src/main/java/org/keycloak/locale/DefaultLocaleSelectorProvider.java#L43-L62
